pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -340,7 +340,7 @@ def should_rank_save_checkpoint(self) -> bool:
     def register_plugins(cls, plugin_registry: Dict) -> None:
         plugin_registry.register("tpu_spawn_debug", cls, description="TPUSpawn Plugin with `debug` as True", debug=True)
 
-    @TPUSpawnPlugin.checkpoint_io.setter
+    @DDPSpawnStrategy.checkpoint_io.setter
```

Comment:
is this intentional?

Reply:
Yeah, otherwise will complain DDPSpawnPlugin doesn't exist

Reply:
This goes beyond my understanding. @carmocca how do these setters even work like this?

Reply:
@awaelchli I think it's because TPU_spawn is calling super.__init__ to set the checkpointer_io, and DDPspawning is the super class